torch/_inductor/codegen/cpp.py

```diff
@@ -472,6 +472,14 @@ def update_on_args(self, name, args, kwargs):
                 self._set_dependent_itervars(args[0])
             if any(arg.is_vec for arg in args if isinstance(arg, CppCSEVariable)):
                 self.is_vec = True
+        # NOTE [dtype of CppCSEVariable]
+        # Deciding dtype according to the current optimization context is not
+        # always accurate since the dtypes are initialized during dtype propagation
+        # at the beginning of the codegen. It is possible that some ops are invoked
+        # during the codegen of the current op and take different dtypes from the
+        # current op.
+        # TODO(jgong5): A more accurate way of deciding the dtype of the variables is to
+        # propagate the dtypes here inside `update_on_args`.
```

Comment:
Should `update_on_args` firstly be invoked at scalar kernel codegen? So, it means we can only get the dtype information afterwards.